src/train.py

```diff
@@ -0,0 +1,314 @@
+                    wandb.log({"recall": recall_score(ground_truth, predictions, zero_division=0)})
+                    wandb.log({"f1": f1_score(ground_truth, predictions, pos_label=1, zero_division=0)})
+
+                model.train()
+
+    pbar.close()
+    if save_to is not None:
+        torch.save(model.state_dict(), save_to)
+
+
+class BceLossForTokenClassification:
+    def __init__(self):
+        self.bce_loss = nn.BCEWithLogitsLoss(reduction="none")
+
+    def __call__(self, outputs: torch.Tensor, labels: torch.Tensor):
+        outputs = outputs.view(-1)
+        labels = labels.view(-1).float()
+        mask = (labels >= 0).float()
+        num_tokens = int(torch.sum(mask))
+        loss = self.bce_loss(outputs, labels) * mask
+        return torch.sum(loss) / num_tokens
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    num = 8
+    parser.add_argument(
+        "--data",
+        type=str,
+        default=f"resources/data/noisy_dataset.jsonl",
+    )
+    parser.add_argument("--tokenizer", type=str, default=f"tokenizer")
+    parser.add_argument("--max-seq-len", type=int, default=512)
+    parser.add_argument("--max-slices-count-per-word", type=int, default=9)
+    parser.add_argument("--font-size", type=int, default=15)
+    parser.add_argument("--window-size", type=int, default=30)
+    parser.add_argument("--stride", type=int, default=5)
+    parser.add_argument("--epochs", type=int, default=10)
+    parser.add_argument("--test-size", type=float, default=0.1)
+    parser.add_argument("--val-size", type=float, default=0.1)
+    parser.add_argument("--random-state", type=int, default=21)
+    parser.add_argument("--log-every", type=int, default=1000)
+    parser.add_argument("--emb-size", type=int, default=768)
+    parser.add_argument("--num-workers", type=int, default=1)
+    parser.add_argument("--save-to", type=str, default=None)
+    parser.add_argument("--font", type=str, default="fonts/NotoSans.ttf")
+    parser.add_argument("--out-channels", type=int, default=256)
+    parser.add_argument("--nhead", type=int, default=12)
+    parser.add_argument("--beta1", type=float, default=0.9)
+    parser.add_argument("--beta2", type=float, default=0.999)
+    parser.add_argument("--sl", action="store_true")
+    parser.add_argument("--test-data", type=str, default=None)
+    parser.add_argument("--val", action="store_true")
+
+    parser.add_argument("--dropout", type=float, default=0.5)
+    parser.add_argument("--device", type=str)
+    parser.add_argument("--warmup", type=int, default=1000)
+    parser.add_argument("--batch-size", type=int, default=32)
+    parser.add_argument("--kernel-size", type=int, default=3)
+    parser.add_argument("--num-layers", type=int, default=1)
+    parser.add_argument("--lr", type=float, default=5e-5)
+
+    args = parser.parse_args()
+
+    wandb.init(project="visual-text", entity="borisshapa", config=vars(args))
```

Comment:
Logic from here and to the end of the script should be placed inside the `main` function. This would make it easy to run model training from notebooks (Colab). Users will import the `main` function and call it with necessary arguments.